cmake/HPX_AddDefinitions.cmake

```diff
@@ -80,7 +80,21 @@ function(write_config_defines_file)
 
   set(hpx_config_defines "\n")
   foreach(def ${DEFINITIONS_VAR})
-    set(hpx_config_defines "${hpx_config_defines}#define ${def}\n")#"
+    # C++17 specific variable
+    string(FIND ${def} "HAVE_CXX17" _pos)
+    if(NOT ${_pos} EQUAL -1)
+      set(hpx_config_defines
+         "${hpx_config_defines}#if __cplusplus >= 201500\n#define ${def}\n#endif\n")
```

Comment:
This is MSVC specific, right?

Reply:
No it isn't. It's gcc specific - normally `__cplusplus == 201703` for C++17 (iirc). I used this to decide on a constant: https://renenyffenegger.ch/notes/development/languages/C-C-plus-plus/preprocessor/macros/predefined/__cplusplus.